doc/proxy-metrics.md

```diff
@@ -0,0 +1,71 @@
++++
+title = "Proxy Metrics"
+docpage = true
+[menu.docs]
+  parent = "proxy-metrics"
++++
+
+The Conduit proxy exposes metrics that describe the traffic flowing through the
+proxy.  These metrics are exposed in Prometheus format on the proxy control port
+(4191 by default) at the `/metrics` path.  The following metrics will be
+exposed:
+
+### `request_total`
+
+A counter of the number of requests the proxy has received.  This is incremented
+when the request stream begins.
+
+### `request_duration_ms`
+
+A histogram of the duration of a request.  This is measured from when the
+request headers are received to when the request stream has completed.
+
+### `response_total`
+
+A counter of the number of responses the proxy has received.  This is
+incremeneted when the response stream ends.
+
+### `response_duration_ms`
+
+A histogram of the duration of a response.  This is measured from when the
+response headers are received to when the response stream has completed.
+
+### `response_latency_ms`
+
+A histogram of the total latency of a response.  This is measured from when the
+request headers are received to when the response stream has completed.
+
+## Labels
+
+Each of these metrics has the following labels:
+
+* `direction`: `inbound` if the request originated from outside of the pod,
+               `outbound` if the request originated from inside of the pod.
+* `deployment`: The deployment that the pod belongs to (if applicable).
+* `job`: The job that the pod belongs to (if applicable).
+* `replica_set`: The replica set that the pod belongs to (if applicable).
+* `daemon_set`: The daemon set that the pod belongs to (if applicable).
+* `replication_controller`: The replication controller that the pod belongs to 
```

Comment:
What do you think about using  a `k8s_` prefix?

Also, this raises a question: how many metrics does a proxy typically expose? How much overhead are we adding with these labels? Should we be using shorter names? Or should we prioritize labeling in the collector?

What do you think about using (k8s standard) abbreviations? `k8s_rc` seems fine as compared to `replication_controller`, for instance.

Reply:
>  Should we be using shorter names?

If we're that concerned about wire size of the exported metrics, should we also exclude the HELP comments or add an option to turn them off? The current implementation includes these messages.

Reply:
Ideally those would be omitted unless explicitly requested (i.e. with a `?help`)

Reply:
Before we begin optimizing for shorter label names and wire size, we should measure what our expected wire size is, and at what thresholds it impacts performance.

Wrt. a `?help` param, that would move us away from a best practices Prometheus deployment to more of a one-off. I'm not aware of any system that exports metrics in that way. More info: https://prometheus.io/docs/instrumenting/writing_exporters/#help-strings

Reply:
these labels come from the `CONDUIT_PROMETHEUS_LABELS` env variable which is set by conduit inject.  So the proxy is somewhat agnostic to them, if that's your concern.  I don't really see the value in signaling that these labels are k8s related in their name but this is easy to change by updating conduit inject.

Reply:
> I don't really see the value in signaling that these labels are k8s related in their name

terms like _job_ are widely used in other contexts

> but this is easy to change by updating conduit inject.

... and all of the queries that read from them.